File: src/main/java/com/muxiu1997/sharewhereiam/command/CommandWaypointToggleTempBeacon.java
package com.muxiu1997.sharewhereiam.command;

import com.muxiu1997.sharewhereiam.command.base.CommandWaypointBase;
import com.muxiu1997.sharewhereiam.journeymap.WaypointManager;
import journeymap.client.model.Waypoint;
import net.minecraft.command.CommandException;
import net.minecraft.command.ICommandSender;

public class CommandWaypointToggleTempBeacon extends CommandWaypointBase {

    public static CommandWaypointToggleTempBeacon INSTANCE = new CommandWaypointToggleTempBeacon();

    public CommandWaypointToggleTempBeacon() {
        super("toggletempbeacon");
    }

    @Override
    public void processCommand(ICommandSender sender, String[] args) throws CommandException {
        ensureArgsLength(args, 0, 1);
        if (args.length == 0) {
            WaypointManager.instance.clearActiveWaypoint();
            return;
        }
        Waypoint waypoint = parseWaypoint(args[0]);
        WaypointManager.instance.toggleActiveWaypoint(waypoint);
    }
}

File: src/main/kotlin/com/muxiu1997/sharewhereiam/command/CommandWaypointToggleTempBeacon.kt
package com.muxiu1997.sharewhereiam.command

import com.muxiu1997.sharewhereiam.command.base.CommandWaypointBase
import com.muxiu1997.sharewhereiam.integration.journeymap.WaypointManager
import com.muxiu1997.sharewhereiam.localization.Lang
import net.minecraft.command.CommandException
import net.minecraft.command.ICommandSender

object CommandWaypointToggleTempBeacon : CommandWaypointBase("toggletempbeacon") {
    override val commandUsage = Lang.TOGGLE_TEMP_BEACON_USAGE.key

    @Throws(CommandException::class)
    override fun processCommand(sender: ICommandSender, args: Array<String>) {
        ensureArgsLength(args, 0, 1)
        if (args.isEmpty()) {
            WaypointManager.clearActiveWaypoint()
            return
        }
        val waypoint = parseWaypoint(args[0])
        WaypointManager.toggleActiveWaypoint(waypoint)
    }
}
